### PR TITLE
Adds Number Threshold, Count Contours, and Crop Operations

### PR DESCRIPTION
**Pull request by JLLeitschuh**
Rewrite of @MeRPG's PR.

Closes #529

### CODE REVIEW

**Comment by SamCarlberg** [comment]
Why do we need an operation to count contours? Why not just add a `size` field to the existing publishing operations?

**Comment by AustinShalit** [comment]
Number Threshold and Count Contours would be good examples of python operations but should not be added as GRIP operations.

I am ok with the crop operation.

File: core/src/main/java/edu/wpi/grip/core/operations/Operations.java [hunk 1]
```diff
@@ -67,6 +76,52 @@
...
+                        OperationDescription.builder()
+                                .name("Crop")
+                                .summary("Crop an image")
+                                .icon(Icon.iconStream("grip"))
+                                .build(),
+                        templateFactory.create(
+                                SocketHints.Inputs.createMatSocketHint("src", false),
+                                SocketHints.Inputs.createPointSocketHint("p1", false),
+                                SocketHints.Inputs.createPointSocketHint("p2", false),
+                                SocketHints.Inputs.createMatSocketHint("dst", true),
```
Comment (PatrickPenguinTurtle) [inline core/src/main/java/edu/wpi/grip/core/operations/Operations.java]: Doesn't this make the dst an Input?

Comment (SamCarlberg) [inline core/src/main/java/edu/wpi/grip/core/operations/Operations.java]: No. But the Crop operation should be in its own class file like every other custom operation.

**Comment by SamCarlberg** [comment]
Rejected. Custom operations should be in their own class files.

File: core/src/main/java/edu/wpi/grip/core/operations/Operations.java (hunk 1 above)
Comment (PatrickPenguinTurtle) [inline core/src/main/java/edu/wpi/grip/core/operations/Operations.java]: Wait... SocketHints.Inputs.createMatSocketHint() can be used to create a socket hint used for an output? Isn't that a bit misleading?

Comment (SamCarlberg) [inline core/src/main/java/edu/wpi/grip/core/operations/Operations.java]: Yep. `SocketHints` needs an overhaul (and Javadocs).

Comment (JLLeitschuh) [inline core/src/main/java/edu/wpi/grip/core/operations/Operations.java]: How so? What are you thinking about this? I agree but I want to know what your thoughts are.
Open an issue and we can discuss that there.

**Comment by AustinShalit** [comment]
I am going to close this PR because it does not look like this is going to happen in the near future. Please reopen if you have new information.

**Pull request closed without merging**